auth/api/grpc/endpoint_test.go

```diff
@@ -218,63 +256,92 @@ func TestAuthorize(t *testing.T) {
-			err:      nil,
-			code:     codes.InvalidArgument,
+		},
+		{
+			desc:        "authorize user with unauthorized token",
+			token:       inValidToken,
+			subject:     id,
+			subjecttype: usersType,
+			object:      authoritiesObj,
+			objecttype:  usersType,
+			relation:    memberRelation,
+			permission:  adminpermission,
+			ar:          &magistrala.AuthorizeRes{Authorized: false},
+			err:         svcerr.ErrAuthorization,
+		},
+		{
+			desc:        "authorize user with empty subject",
+			token:       validToken,
+			subject:     "",
+			subjecttype: usersType,
+			object:      authoritiesObj,
+			objecttype:  usersType,
+			relation:    memberRelation,
+			permission:  adminpermission,
+			ar:          &magistrala.AuthorizeRes{Authorized: false},
+			err:         apiutil.ErrMissingPolicySub,
+		},
+		{
+			desc:        "authorize user with empty subject type",
+			token:       validToken,
+			subject:     id,
+			subjecttype: "",
+			object:      authoritiesObj,
+			objecttype:  usersType,
+			relation:    memberRelation,
+			permission:  adminpermission,
+			ar:          &magistrala.AuthorizeRes{Authorized: false},
+			err:         apiutil.ErrMissingPolicySub,
+		},
+		{
+			desc:        "authorize user with empty object",
+			token:       validToken,
+			subject:     id,
+			subjecttype: usersType,
+			object:      "",
+			objecttype:  usersType,
+			relation:    memberRelation,
+			permission:  adminpermission,
+			ar:          &magistrala.AuthorizeRes{Authorized: false},
+			err:         apiutil.ErrMissingPolicyObj,
+		},
+		{
+			desc:        "authorize user with empty object type",
+			token:       validToken,
+			subject:     id,
+			subjecttype: usersType,
+			object:      authoritiesObj,
+			objecttype:  "",
+			relation:    memberRelation,
+			permission:  adminpermission,
+			ar:          &magistrala.AuthorizeRes{Authorized: false},
+			err:         apiutil.ErrMissingPolicyObj,
+		},
+		{
+			desc:        "authorize user with empty permission",
+			token:       validToken,
+			subject:     id,
+			subjecttype: usersType,
+			object:      authoritiesObj,
+			objecttype:  usersType,
+			relation:    memberRelation,
+			permission:  "",
+			ar:          &magistrala.AuthorizeRes{Authorized: false},
+			err:         apiutil.ErrMalformedPolicyPer,
 		},
 	}
 	for _, tc := range cases {
-		repocall := prepo.On("CheckPolicy", mock.Anything, mock.Anything).Return(nil)
-		ar, _ := client.Authorize(context.Background(), &magistrala.AuthorizeReq{Subject: tc.subject, SubjectType: tc.subjecttype, Object: tc.object, ObjectType: tc.objecttype, Relation: tc.relation, Permission: tc.permission})
+		repocall := svc.On("Authorize", mock.Anything, mock.Anything).Return(tc.err)
+		ar, err := client.Authorize(context.Background(), &magistrala.AuthorizeReq{Subject: tc.subject, SubjectType: tc.subjecttype, Object: tc.object, ObjectType: tc.objecttype, Relation: tc.relation, Permission: tc.permission})
```

Comment:
Combine them so that you can use 
```suggestion
		ar, err := client.Authorize(context.Background(), tc.authReq)
```

Reply:
Same as other cases